sdk/core/Azure.Core/src/Shared/OperationInternalBase.cs

```diff
@@ -188,20 +192,34 @@ private async ValueTask<Response> WaitForCompletionResponseAsync(bool async, Tim
                 return lockOrValue.Value;
             }
 
-            var poller = new OperationPoller(_fallbackStrategy);
-            var response = async
-                ? await poller.WaitForCompletionResponseAsync(this, pollingInterval, cancellationToken).ConfigureAwait(false)
-                : poller.WaitForCompletionResponse(this, pollingInterval, cancellationToken);
+            using var scope = CreateScope(_waitForCompletionScopeName);
+            try
+            {
+                var poller = new OperationPoller(_fallbackStrategy);
+                var response = async
+                    ? await poller.WaitForCompletionResponseAsync(this, pollingInterval, cancellationToken).ConfigureAwait(false)
+                    : poller.WaitForCompletionResponse(this, pollingInterval, cancellationToken);
 
-            lockOrValue.SetValue(response);
-            return response;
+                lockOrValue.SetValue(response);
+                return response;
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
         }
 
         protected abstract ValueTask<Response> UpdateStatusAsync(bool async, CancellationToken cancellationToken);
 
-        protected DiagnosticScope CreateScope()
+        protected DiagnosticScope CreateScope(string scopeName)
         {
-            DiagnosticScope scope = _diagnostics.CreateScope(_updateStatusScopeName);
+            if (scopeName != _updateStatusScopeName && scopeName != _waitForCompletionScopeName)
+            {
+                throw new NotSupportedException($"The scope name is not supported. Allowed value: {_updateStatusScopeName}, {_waitForCompletionScopeName}");
+            }
```

Comment:
Why do we need add this check? CreateScope is a function to create a scope, it is not a dedicate function to only create UpdateStatusScope or WaitForCompletionScope.  When you call this function you need to pass the right scope name.

Reply:
Updated. But just want to confirm: we don't expect RP who might customize the `operation` will pass in an invalid scope name, right?

Reply:
I don't think we validate this for scopes elsewhere -- would you be able to confirm this?

@JoshLove-msft/@christothes/@AlexanderSher, is it correct that we validate diagnostic scope names only through tests?  Could you point @pshao25 to documentation/examples of how we do this?

Reply:
Yes, when you instrument your client as part of recorded tests, we add an interceptor that will validate that a scope is created that matches the name of the method. There is a gap in that if you are using the ForwardClientCalls attribute and creating the scope from a different internal method, the name is not validated.